staff_features/resources/step_definitions/resource_tree_navigate.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+Given 'a Resource with two Archival Objects has been created' do
+  visit "#{STAFF_URL}/resources/new"
+
+  fill_in 'resource_title_', with: "Resource #{@uuid}"
+  fill_in 'resource_id_0_', with: "Resource #{@uuid}"
+  select 'Class', from: 'resource_level_'
+  element = find('#resource_lang_materials__0__language_and_script__language_')
+  element.send_keys('AU')
+  element.send_keys(:tab)
+
+  select 'Single', from: 'resource_dates__0__date_type_'
+  fill_in 'resource_dates__0__begin_', with: '2024'
+
+  fill_in 'resource_extents__0__number_', with: '10'
+  select 'Cassettes', from: 'resource_extents__0__extent_type_'
+
+  element = find('#resource_finding_aid_language_')
+  element.send_keys('ENG')
+  element.send_keys(:tab)
+
+  element = find('#resource_finding_aid_script_')
+  element.send_keys('Latin')
+  element.send_keys(:tab)
+
+  find('button', text: 'Save Resource', match: :first).click
+
+  wait_for_ajax
+  expect(page).to have_text "Resource Resource #{@uuid} created"
+
+  url_parts = current_url.split('/')
+  url_parts.pop
+  @resource_id = url_parts.pop
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 1 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 1 #{@uuid} on Resource Resource #{@uuid} created"
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 2 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 2 #{@uuid} created as child of Archival Object 1 #{@uuid} on Resource Resource #{@uuid}"
+  wait_for_ajax
+end
+
+Given 'the Resource is opened in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+When 'the user opens the Resource in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+Then 'the Resource is displayed as the top level of the navigation tree' do
+  rows = all('#tree-container .table .table-row', visible: :all)
+
+  expect(rows.length).to eq 4
+  expect(rows[0].text).to include "Resource #{@uuid}"
+end
+
+Then 'the tree includes the two Archival Objects' do
+  rows = all('#tree-container .table .table-row')
+
+  expect(rows.length).to eq 2
+  expect(rows[2].text).to include "Archival Object 1 #{@uuid}"
+end
+
+Then 'the Resource is highlighted in the tree' do
```

Comment:
Very nitpicky, but this doesn't actually test that the resource is highlighted, only that it has a class that presumably means it would be.  Maybe change the title to 'the Resource class is current' ?

Reply:
Personally, I agree, however, these tests heavily depend on the UI. The term `highlighted` also has a meaning only in the UI. There is no other way to actually test that something is highlighted. Even though we are currently writing these tests, in the future, it is supposed to be that non-technical people should be able to write Gherkin and describe features. This means that the Gherkin syntax has to be agnostic of technical details like HTML selectors, tags etc. I suggest that we should probably completely remove this step as it adds no value to the actual test. What do you think?

Reply:
Well, there could be value in testing that the system did take note of what it has to highlight, even if it's not possible to actually test that it _is_ highlighted.  I think 'd vote for keeping the test over worrying about making sure the title is 100% accurate. It is just semantics at this point but you could change the title of the test to "the Resource should be highlighted in the tree" and it would be mostly representative. But I'll defer to @brianzelip on this one.

Reply:
@balmas @blacksmith-welder It is a commonly practiced convention in ArchivesSpace feature tests to test for the presence of `.current` on elements expected to be highlighted. I think there is value in the step since it tests intentional system feedback for the user. There are many places in the UI where we want to ensure we are providing such system feedback. At the moment I'm not sure of a better option than the approach this step takes. I think this test should be accepted as is.